lib/iris/fileformats/netcdf/loader.py

```diff
@@ -655,6 +659,17 @@ def load_cubes(file_sources, callback=None, constraints=None):
 
                 yield cube
 
+    # TODO: consider propagating to other loaders in future.
+    #  Could not apply at a higher level as user support and ncdata both access
+    #  this lower-level loader.
+    if _profile_load_problems() != load_problems_original:
+        # TODO: should this print iris.loading.LOAD_PROBLEMS?
+        message = (
+            "Not all file objects were parsed correctly. See "
+            "iris.loading.LOAD_PROBLEMS for details."
+        )
+        warnings.warn(message, category=IrisLoadWarning)
```

Comment:
📢

Reply:
Offline, we agreed:

- I am going to move the warning up to `_load_collection`. Anyone directly calling lower level operations is a super-user and can cope without the warning. Or they are using ncdata and @pp-mo is hoping to change how ncdata interacts with the Iris loader
- There is no need to do any automatic clearing, as this makes sweeping assumptions about user needs. However it should be well advertised that `LOAD_PROBLEMS` _can_ be cleared, perhaps when making a specialised class object - see https://github.com/SciTools/iris/pull/6338#discussion_r1985128826

Reply:
>Offline, we agreed:
>...

The refactor in bc98b2f72362998320567d1f4d86a823c55b3009 addresses both these bullets:

- Having a dedicated `LoadProblems` class allowed me to address warnings directly in there, so I no longer need to worry about where to raise the warning. Note that this meant several more tests in `tests/unit/fileformats/nc_load_rules/actions/` raised warnings where they weren't raised before.
  https://github.com/SciTools/iris/blob/5a7f68c10fd6f468457bb768e1064ad586aca91e/lib/iris/loading.py#L579-L587
- The new `LoadProblems` class implements the suggested `reset()` method, which includes a docstring, and a parameter to select a specific file to 'clear'.